In [1]:
import pandas as pd

In [10]:
LoanData=pd.read_csv('Loan.csv')
LoanData.head()

gender married   ch  income  loanamt status
0   Male      No  1.0    5849      NaN      Y
1   Male     Yes  1.0    4583    128.0      N
2   Male     Yes  1.0    3000     66.0      Y
3   Male     Yes  1.0    2583    120.0      Y
4   Male      No  1.0    6000    141.0      Y

In [4]:
LoanPrep=LoanData.copy()

In [5]:
LoanPrep.isnull().sum()

gender     13
married     3
ch         50
income      0
loanamt    22
status      0
dtype: int64

In [7]:
LoanPrep=LoanPrep.dropna()

In [8]:
LoanPrep.isnull().sum()

gender     0
married    0
ch         0
income     0
loanamt    0
status     0
dtype: int64

In [9]:
LoanPrep=LoanPrep.drop(['gender'],axis=1)

In [17]:
LoanPrep.dtypes

married     object
ch         float64
income     float64
loanamt    float64
status      object
dtype: object

In [20]:
LoanPrep=pd.get_dummies(LoanPrep,drop_first=True)
LoanPrep.head()

ch    income   loanamt  married_Yes  status_Y
1  1.0 -0.128073 -0.194250            1         0
2  1.0 -0.392077 -0.971015            1         1
3  1.0 -0.461621 -0.294478            1         1
4  1.0  0.108246 -0.031380            0         1
5  1.0  0.011017  1.547205            1         1

In [21]:
from sklearn.preprocessing import StandardScaler
scalar_=StandardScaler()

In [22]:
LoanPrep['income']=scalar_.fit_transform(LoanPrep[['income']])

In [23]:
LoanPrep['loanamt']=scalar_.fit_transform(LoanPrep[['loanamt']]) 
# normalization

In [24]:
LoanPrep

ch    income   loanamt  married_Yes  status_Y
1    1.0 -0.128073 -0.194250            1         0
2    1.0 -0.392077 -0.971015            1         1
3    1.0 -0.461621 -0.294478            1         1
4    1.0  0.108246 -0.031380            0         1
5    1.0  0.011017  1.547205            1         1
..   ...       ...       ...          ...       ...
609  1.0 -0.408754 -0.908372            0         1
610  1.0 -0.207624 -1.296754            1         1
611  1.0  0.453802  1.371807            1         1
612  1.0  0.372249  0.544929            1         1
613  0.0 -0.128073 -0.131608            0         0

[529 rows x 5 columns]

In [25]:
Y=LoanPrep[['status_Y']]
X=LoanPrep.drop(['status_Y'],axis=1)

In [27]:
X

ch    income   loanamt  married_Yes
1    1.0 -0.128073 -0.194250            1
2    1.0 -0.392077 -0.971015            1
3    1.0 -0.461621 -0.294478            1
4    1.0  0.108246 -0.031380            0
5    1.0  0.011017  1.547205            1
..   ...       ...       ...          ...
609  1.0 -0.408754 -0.908372            0
610  1.0 -0.207624 -1.296754            1
611  1.0  0.453802  1.371807            1
612  1.0  0.372249  0.544929            1
613  0.0 -0.128073 -0.131608            0

[529 rows x 4 columns]

In [28]:
from sklearn.model_selection import train_test_split
X_train,X_test,Y_train,Y_test=train_test_split(X,Y,test_size=0.3,random_state=1234,stratify=Y)

In [30]:
from sklearn.linear_model import LogisticRegression

In [31]:
lr=LogisticRegression()
lr.fit(X_train,Y_train)

C:\anaconda\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=100,
                   multi_class='auto', n_jobs=None, penalty='l2',
                   random_state=None, solver='lbfgs', tol=0.0001, verbose=0,
                   warm_start=False)

In [32]:
Y_predict=lr.predict(X_test)

In [33]:
Y_predict

array([0, 0, 1, 1, 1, 0, 1, 1, 0, 1, 1, 0, 1, 0, 0, 1, 1, 1, 1, 0, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 1, 1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1,
       0, 0, 1, 1, 1, 1, 0, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 0,
       1, 1, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1, 1,
       1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1,
       1, 1, 1, 1, 1], dtype=uint8)

In [34]:
Y_test

status_Y
373         0
464         0
3           1
262         0
401         0
..        ...
360         0
71          1
96          1
388         1
270         1

[159 rows x 1 columns]

In [35]:
from sklearn.metrics import confusion_matrix
cm=confusion_matrix(Y_predict,Y_test)

In [36]:
lr.score(X_test,Y_test)

0.8616352201257862

86% accurate in predicting the true positive and negative.only 14% has false predictions